queries.sql

```diff
@@ -1,27 +1,39 @@
-SELECT animals.name from animals INNER JOIN owners on animals.owner_id=owners.id
-WHERE owners.full_name='Melody Pond' ORDER BY animals.name;
+SELECT animals.name, visits.visit_date FROM visits INNER JOIN animals ON animal_id = animals.id
+INNER JOIN vets ON vet_id=vets.id WHERE vets.name='William Tatcher' ORDER BY visits.visit_date DESC LIMIT 1;
 
-SELECT animals.name FROM animals INNER JOIN species ON animals.species_id = species.id
-WHERE species.name='Pokemon' ORDER BY animals.name;
+SELECT COUNT(DISTINCT animals.name) FROM visits INNER JOIN animals ON animal_id = animals.id
+INNER JOIN vets ON vet_id=vets.id WHERE vets.name='Stephanie Mendez';
 
-SELECT owners.full_name, animals.name FROM owners LEFT JOIN animals ON owners.id = animals.owner_id
-ORDER BY owners.full_name;
+SELECT vets.name, species.name FROM vets LEFT JOIN specializations ON vets.id = vet_id
+LEFT JOIN species ON species_id=species.id ORDER BY vets.name;
 
-SELECT species.name, COUNT(animals.name) AS count FROM animals  
-INNER JOIN species ON animals.species_id = species.id GROUP BY species.id
-ORDER BY count DESC;
-
-SELECT animals.name FROM animals
-INNER JOIN owners ON animals.owner_id = owners.id 
-INNER JOIN species ON animals.species_id = species.id
-WHERE species.name='Digimon' AND owners.full_name='Jennifer Orwell'
+SELECT animals.name FROM visits INNER JOIN animals ON animal_id= animals.id
+INNER JOIN vets ON vet_id=vets.id
+WHERE vets.name='Stephanie Mendez' AND visit_date BETWEEN '2020-04-01' AND '2020-08-30'
 ORDER BY animals.name;
 
-SELECT animals.name FROM animals
-INNER JOIN owners ON animals.owner_id = owners.id 
-WHERE owners.full_name='Dean Winchester' AND escape_attempts=0
-ORDER BY animals.name;
+SELECT animals.name, COUNT(animals.name) AS number_visits FROM visits INNER JOIN animals ON animal_id= animals.id 
+GROUP BY animals.name ORDER BY number_visits DESC LIMIT 1;
+
+
+SELECT animals.name, visit_date  FROM visits INNER JOIN animals ON animal_id= animals.id
+INNER JOIN vets ON vet_id=vets.id
+WHERE vets.name='Maisy Smith'
+ORDER BY visit_date ASC LIMIT 1;
+
+SELECT animals.name, animals.date_of_birth, animals.escape_attempts, animals.neutered, animals.weight_kg, 
+vets.name, vets.age, vets.date_of_graduation
+visit_date  FROM visits INNER JOIN animals ON animal_id= animals.id
+INNER JOIN vets ON vet_id=vets.id
+ORDER BY visit_date DESC LIMIT 1;
```

Comment:
- [ ] (Optional) I am making this optional because the query is correct but you have a typo. You did not include a comma(,) after `vets.date_of_graduation` just before `visit_date`, so it returned a wrong result. Kindly make the change or adopt my suggestion. 👍 

- [ ] (Optional) There is a shorthand for selecting all columns. it is ` SELECT animals.*, vets.*, visits.visit_date`

```suggestion
SELECT animals.name, animals.date_of_birth, animals.escape_attempts, animals.neutered, animals.weight_kg, 
vets.name, vets.age, vets.date_of_graduation,
visit_date  FROM visits INNER JOIN animals ON animal_id= animals.id
INNER JOIN vets ON vet_id=vets.id
ORDER BY visit_date DESC LIMIT 1;
```